omnisend-for-ninja-forms/includes/Mapper/class-formfieldsmapper.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Form fields mapper
+ *
+ * @package OmnisendNinjaFormsPlugin
+ */
+
+declare(strict_types=1);
+
+namespace Omnisend\NinjaFormsAddon\Mapper;
+
+use Omnisend\NinjaFormsAddon\Actions\OmnisendAddOnAction;
+use Omnisend\NinjaFormsAddon\Provider\OmnisendActionSettingsProvider;
+
+/**
+ * Class FormFieldsMapper
+ */
+class FormFieldsMapper {
+
+	/**
+	 * Get field mappings.
+	 *
+	 * @param array $action_settings Action settings.
+	 * @param array $form_fields     Form fields.
+	 * @return array Field mappings.
+	 */
+	public function get_field_mappings( array $action_settings, array $form_fields ): array {
+		$field_mappings = array();
+
+		foreach ( OmnisendAddOnAction::OMNISEND_FIELDS as $key => $label ) {
+			if ( ! isset( $action_settings[ $key ] ) ) {
+				continue;
+			}
+
+			if ( 'default' === $action_settings[ $key ] || '-' === $action_settings[ $key ] ) {
+				continue;
+			}
+
+			$field_mappings[ $key ] = $action_settings[ $key ];
+		}
+
+		$values         = array();
+		$consent_fields = array( OmnisendAddOnAction::EMAIL_CONSENT, OmnisendAddOnAction::PHONE_CONSENT );
+
+		foreach ( $field_mappings as $key => $field ) {
+			foreach ( $form_fields['fields'] as $form_field ) {
+				if ( $form_field['key'] !== $field ) {
+					continue;
+				}
+
+				if ( in_array( $key, $consent_fields, true ) ) {
+					if ( '1' === $form_field['value'] ) {
+						$values[ $key ] = 'subscribed';
+					} else {
+						$values[ $key ] = 'nonSubscribed';
+					}
+
+					continue;
+				}
+
+				if ( OmnisendAddOnAction::BIRTHDAY === $key ) {
+					$values[ $key ] = gmdate( 'Y-m-d', strtotime( $form_field['value'] ) );
```

Comment:
can this fail if the value is not possible to cast to time?

Reply:
It will not fail, but I've added additional checks to make sure.